### PR TITLE
Editable Textfield Angular 7 Component

### PR DESCRIPTION
**Pull request by daanishnasir**
## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: angular/src/lib/editable-textfield/editable-textfield.component.ts [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import { ControlValueAccessor, NgControl } from '@angular/forms';
+
+const cb = () => {};
+@Component({
+  selector: 'cui-editable-textfield',
+  template: `
+    <span
+      class="cui-editable-textfield"
+      [ngClass]="wrapperClasses"
+    >
```
Comment (wzpo) [inline angular/src/lib/editable-textfield/editable-textfield.component.ts]: The span can be removed and bind class to host.

Comment (daanishnasir) [reply angular/src/lib/editable-textfield/editable-textfield.component.ts]: removed and updated to new pattern

File: angular/src/lib/editable-textfield/editable-textfield.module.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+@NgModule({
+  declarations: [EditableTextfieldComponent],
+  imports: [
+    CommonModule,
+    InputModule,
+    FormsModule,
+    ReactiveFormsModule
+  ],
+  exports: [EditableTextfieldComponent]
```
Comment (elenagiere) [inline angular/src/lib/editable-textfield/editable-textfield.module.ts]: I think we are trying to add 
`schemas: [ CUSTOM_ELEMENTS_SCHEMA ]`
to all component modules and test modules

Comment (daanishnasir) [reply angular/src/lib/editable-textfield/editable-textfield.module.ts]: Added, thanks

File: angular/src/lib/editable-textfield/editable-textfield.component.ts [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      <input
+        #inputRef
+        class="cui-input"
+        [ngClass]="inputClasses"
+        [attr.id]="htmlId"
+        [(ngModel)]="value"
+        [placeholder]="placeholder"
+        [readonly]="readOnly"
+        [attr.disabled]="disabled ? '' :null"
```
Comment (bfbiggs) [inline angular/src/lib/editable-textfield/editable-textfield.component.ts]: Space after colon.

File: angular/src/lib/editable-textfield/editable-textfield.component.ts [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+
+  registerOnChange(fn: any) {
+    this.onChangeCallback = fn;
+  }
+
+  registerOnTouched(fn: any) {
+    this.onTouchedCallback = fn;
+  }
+
+  onKeyDown(e){
```
Comment (bfbiggs) [inline angular/src/lib/editable-textfield/editable-textfield.component.ts]: Some functions have space after parens, some don't.

File: angular/src/lib/editable-textfield/examples/editable-textfield-examples.module.ts [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { NgModule, CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { EditableTextfieldDefaultComponent } from './editable-textfield-default.component';
+import { FormsModule, ReactiveFormsModule } from '@angular/forms';
+
+import {EditableTextfieldModule } from '@collab-ui/angular'
```
Comment (bfbiggs) [inline angular/src/lib/editable-textfield/examples/editable-textfield-examples.module.ts]: Space after first bracket.

**Pull request merged**